### PR TITLE
[api] fix, set default columns to just id when not defined

### PR DESCRIPTION
**Pull request by dpgaspar**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Sets the default column to the Model's id for GET, PUT, POST for all `ModelRestApi` when nothing is defined. Will add tests after #8917 gets merged

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by craig-rueda — APPROVED** [review]
LGTM - needs tests

**Comment by dpgaspar** [comment]
Added tests

File: superset/views/base_api.py [hunk 1]
```diff
@@ -82,6 +82,18 @@ class BaseSupersetModelRestApi(ModelRestApi):
         }
     """  # pylint: disable=pointless-string-statement
 
+    def _init_properties(self):
```
Comment (willbarrett) [inline superset/views/base_api.py]: Would it be possible to set the default column list to `[]` (empty array) to have nothing returned by default? Database primary keys can be sensitive under certain circumstances.

Comment (dpgaspar) [reply superset/views/base_api.py]: Everything is possible, on this case would probably involve a somewhat delicate change on FAB. My take is that, this is just a fallback, enough to force everyone to always declare CRUD fields explicitly or else the API will return nothing of value (like just the id's)

Comment (willbarrett) [reply superset/views/base_api.py]: I think being able to default to fully-closed is important. This isn't a blocker for this PR, but it's a change I would really like to see on FAB.

**Review by willbarrett — APPROVED** [review]

**Pull request merged**